library/java/src/io/envoyproxy/envoymobile/engine/AndroidEngineImpl.java

```diff
@@ -13,9 +14,8 @@
 
   // Private helper class used by the load method to ensure the native library and its
   // dependencies are loaded and initialized at most once.
-  private AndroidEngineImpl(Context context) {
-    System.loadLibrary("envoy_jni");
-    initialize((ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE));
+  public AndroidEngineImpl(Context context) {
+    initialize((ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
```

Comment:
We have a few options here and this is something we can figure out separate from this PR:

If `c-ares` is initialized, we can:
1. Flip a global static bit in the platform layer to signal all other `AndroidEngineImpl` instances to skip this initialization call
2. Flip a global static bit in the native layer
3. Have a method which exposes if `c-ares` is initialized (we have a native method already)